mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLHostOps.td

```diff
@@ -0,0 +1,42 @@
+//===-- SYCLHostOps.td - SYCL dialect operation definitions *- tablegen -*-===//
+//
+// MLIR-SYCL is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This is the operation definition file for SYCL dialect operations involving
+// host code.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SYCL_HOST_OPS
+#define SYCL_HOST_OPS
+
+include "mlir/Dialect/LLVMIR/LLVMOpBase.td"
+
+////////////////////////////////////////////////////////////////////////////////
+// BASE DECLARATIONS
+////////////////////////////////////////////////////////////////////////////////
+
+class SYCL_HostOp<string mnemonic, list<Trait> traits = []> :
+    SYCL_Op<"host." # mnemonic, traits>;
+
+////////////////////////////////////////////////////////////////////////////////
+// OPERATIONS
+////////////////////////////////////////////////////////////////////////////////
+
+def SYCLHostConstructorOp : SYCL_HostOp<"constructor"> {
```

Comment:
Can we document the rationale for this operation. Something similar to what you have in the PR description.